Move off of UIKit SPI: -[UIScrollView _is(Vertical|Horizontal)Bouncing]

whsieh opened the pull request:
#### 2f74816e605d1c2539f2613ca928bda62dd21c3c
<pre>
Move off of UIKit SPI: -[UIScrollView _is(Vertical|Horizontal)Bouncing]
<a href="https://bugs.webkit.org/show_bug.cgi?id=262013">https://bugs.webkit.org/show_bug.cgi?id=262013</a>
rdar://112473874

Reviewed by Aditya Keerthi.

Stop using `-_isVerticalBouncing` and `-_isHorizontalBouncing` on `UIScrollView`; as an alternative,
add a new category helper method, `-[UIScrollView _wk_isScrolledBeyondExtents]`, that returns
whether or not the current content offset is scrolled beyond the minimum or maximum extents of the
scroll view, taking the `-adjustedContentInset` into account.

* Source/WebKit/Platform/spi/ios/UIKitSPI.h:
* Source/WebKit/UIProcess/ios/UIKitUtilities.h:
* Source/WebKit/UIProcess/ios/UIKitUtilities.mm:
(-[UIScrollView _wk_isScrolledBeyondExtents]):
* Source/WebKit/UIProcess/ios/WKScrollView.mm:
(-[WKScrollView _setContentSizePreservingContentOffsetDuringRubberband:]):
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/cocoa/TestRunnerWKWebView.mm:
* Tools/WebKitTestRunner/ios/UIKitSPIForTesting.h: Copied from Source/WebKit/UIProcess/ios/UIKitUtilities.h.

Continue using `-_isVerticalBouncing` and `-_isHorizontalBouncing` in WebKitTestRunner harness logic
to determine when scrolling/zooming animations are complete. We also add an SPI header, specific to
WebKitTestRunner for now, to house these new forward declarations and SPI include points. I&apos;ll move
various existing testing-only declarations and includes in UIKitSPI.h into this header in a
subsequent patch.

Canonical link: <a href="https://commits.webkit.org/268376@main">https://commits.webkit.org/268376@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cead75a7ee66d5280dfbe615704cdff35ea4151d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19515 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19934 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20542 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21405 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18251 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/19751 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23197 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20076 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/19854 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19731 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19753 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16960 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22261 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16940 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17748 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24058 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18001 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17923 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22031 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18531 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/15700 "1 failures") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17675 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4667 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22031 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18359 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
> We also add an SPI header, specific to
WebKitTestRunner for now, to house these new forward declarations and SPI include points.

Thanks! This is the first step for https://bugs.webkit.org/show_bug.cgi?id=260510 😄

aprotyas reviewed (commented):
Also, could we not use `_wk_isScrolledBeyondExtents` in WKTR?

whsieh committed:
Move off of UIKit SPI: -[UIScrollView _is(Vertical|Horizontal)Bouncing]
https://bugs.webkit.org/show_bug.cgi?id=262013
rdar://112473874

Reviewed by Aditya Keerthi.

Stop using `-_isVerticalBouncing` and `-_isHorizontalBouncing` on `UIScrollView`; as an alternative,
add a new category helper method, `-[UIScrollView _wk_isScrolledBeyondExtents]`, that returns
whether or not the current content offset is scrolled beyond the minimum or maximum extents of the
scroll view, taking the `-adjustedContentInset` into account.

* Source/WebKit/Platform/spi/ios/UIKitSPI.h:
* Source/WebKit/UIProcess/ios/UIKitUtilities.h:
* Source/WebKit/UIProcess/ios/UIKitUtilities.mm:
(-[UIScrollView _wk_isScrolledBeyondExtents]):
* Source/WebKit/UIProcess/ios/WKScrollView.mm:
(-[WKScrollView _setContentSizePreservingContentOffsetDuringRubberband:]):
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/cocoa/TestRunnerWKWebView.mm:
* Tools/WebKitTestRunner/ios/UIKitSPIForTesting.h: Copied from Source/WebKit/UIProcess/ios/UIKitUtilities.h.

Continue using `-_isVerticalBouncing` and `-_isHorizontalBouncing` in WebKitTestRunner harness logic
to determine when scrolling/zooming animations are complete. We also add an SPI header, specific to
WebKitTestRunner for now, to house these new forward declarations and SPI include points. I'll move
various existing testing-only declarations and includes in UIKitSPI.h into this header in a
subsequent patch.

Canonical link: https://commits.webkit.org/268376@main
